Xamarin.Forms.Core/CornerRadius.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+
+namespace Xamarin.Forms.Core
+{
+	public struct CornerRadius
+	{
+		public double TopLeft { get; set; }
+		public double TopRight { get; set; }
+		public double BottomLeft { get; set; }
+		public double BottomRight { get; set; }
+
+		public CornerRadius(double uniformRadius) : this(uniformRadius, uniformRadius, uniformRadius, uniformRadius)
+		{
+		}
+
+		public CornerRadius(double topLeft, double topRight, double bottomLeft, double bottomRight)
+		{
+			if (topLeft < 0 || double.IsNaN(topLeft))
```

Comment:
don't do value check in the ctor, but when the property is applied.

Reply:
Changed.